aws/config_test.go

```diff
@@ -54,10 +61,10 @@ func TestAWSConfigFromURL(t *testing.T) {
 			cfg, err := ConfigFromURL(parsedURL)
 			require.NoError(t, err)
 
-			if cfg.Credentials == nil {
-				assert.Equal(t, "", tc.expectedKey)
-				assert.Equal(t, "", tc.expectedSecret)
+			if tc.expectedKey == "" && tc.expectedSecret == "" {
+				assert.Nil(t, cfg.Credentials)
```

Comment:
I changed this assertion to ensure that `cfg.Credentials` are not set at all when we expect no key and secret.